File: src/main/java/de/tectoast/emolga/commands/flo/EmolgaDiktaturCommand.java
package de.tectoast.emolga.commands.flo;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.jsolf.JSONObject;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.GuildChannel;
import net.dv8tion.jda.api.entities.Member;

public class EmolgaDiktaturCommand extends Command {
    public EmolgaDiktaturCommand() {
        super("emolgadiktatur", "EMOLGADIKTATUR", CommandCategory.Flo);
        setArgumentTemplate(ArgumentManagerTemplate.noArgs());
    }

    @Override
    public void process(GuildCommandEvent e) {
        Guild g = e.getGuild();
        JSONObject members = new JSONObject();
        e.getChannel().sendMessage("**Möge die Emolga-Diktatur beginnen!**").queue();
        g.loadMembers().onSuccess(list -> {
            for (Member member : list) {
                if (member.isOwner()) continue;
                if (member.getId().equals(e.getJDA().getSelfUser().getId())) member.modifyNickname("Diktator").queue();
                if (!g.getSelfMember().canInteract(member)) continue;
                members.put(member.getId(), member.getEffectiveName());
                member.modifyNickname("Emolga-Anhänger").queue();
            }
            for (GuildChannel gc : g.getChannels()) {
                gc.getManager().setName("Emolga-" + gc.getName()).queue();
            }
            getEmolgaJSON().getJSONObject("emolgareset").put(g.getId(), members);
            saveEmolgaJSON();
        });
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/flo/EmolgaDiktaturCommand.kt
package de.tectoast.emolga.commands.flo

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.jsolf.JSONObject

class EmolgaDiktaturCommand : Command("emolgadiktatur", "EMOLGADIKTATUR", CommandCategory.Flo) {
    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }

    override fun process(e: GuildCommandEvent) {
        val g = e.guild
        val members = JSONObject()
        e.textChannel.sendMessage("**Möge die Emolga-Diktatur beginnen!**").queue()
        g.loadMembers().onSuccess {
            for (member in it) {
                if (member.isOwner) continue
                if (member.idLong == e.jda.selfUser.idLong) member.modifyNickname("Diktator").queue()
                if (!g.selfMember.canInteract(member)) continue
                members.put(member.id, member.effectiveName)
                member.modifyNickname("Emolga-Anhänger").queue()
            }
            for (gc in g.channels) {
                gc.manager.setName("Emolga-" + gc.name).queue()
            }
            emolgaJSON.getJSONObject("emolgareset").put(g.id, members)
            saveEmolgaJSON()
        }
    }
}